Build CFO image directly in OpenShift CI as part of CI run

ChristianZaccaria opened the pull request:
Related issue: https://github.com/project-codeflare/codeflare-operator/issues/359 

## Changes
- Building CFO image from Dockerfile provided.
- Planning to patch env variable in ODH once release is available.

## Thoughts
- Are we patching the env var from CFO repo for now? Will need to patch before building I think.
- `basic-dsc.yaml` should be copied over from distributed-workloads repo to codeflare-operator repo for it to be used.

sutaakar commented:
/pj-rehearse

ChristianZaccaria commented:
/pj-rehearse

ChristianZaccaria commented:
/retest

ChristianZaccaria commented:
/pj-rehearse

ChristianZaccaria commented:
/pj-rehearse

sutaakar commented on ci-operator/config/opendatahub-io/codeflare-operator/opendatahub-io-codeflare-operator-main.yaml:
```diff
@@ -48,19 +51,20 @@ tests:
         cd tests && go test -timeout 60m -v ./integration
+      dependencies:
+      - env: ODH-CODEFLARE-OPERATOR-CONTROLLER-IMAGE
+        name: image-registry.openshift-image-registry.svc:5000/openshift-operators/codeflare-operator
```
```suggestion
        name: codeflare-operator
```

ChristianZaccaria replied:
Yeah I just realized that, good catch thanks!

ChristianZaccaria commented:
/pj-rehearse

ChristianZaccaria commented:
/pj-rehearse

ChristianZaccaria commented:
/pj-rehearse

anishasthana commented:
What's the ratoinale behind this guy?

ChristianZaccaria commented:
> What's the ratoinale behind this guy?

This is part of resolving this issue: https://github.com/project-codeflare/codeflare-operator/issues/359 

(I should've tagged it)

anishasthana commented:
Ah, I see -- we will be building a new image as part of the openshift-ci test run. The PR subject made me think we will use openshift-ci for image builds

ChristianZaccaria commented:
> FYI the DSC CR should look like this (for using main branch, to test specific PR it needs to be adjusted ):
> 
> ```
> apiVersion: datasciencecluster.opendatahub.io/v1
> kind: DataScienceCluster
> metadata:
>   labels:
>     app.kubernetes.io/created-by: opendatahub-operator
>     app.kubernetes.io/instance: default
>     app.kubernetes.io/managed-by: kustomize
>     app.kubernetes.io/name: datasciencecluster
>     app.kubernetes.io/part-of: opendatahub-operator
>   name: example-dsc
> spec:
>   components:
>     codeflare:
>       devFlags:
>         manifests:
>           - uri: 'https://api.github.com/repos/opendatahub-io/codeflare-operator/tarball/main'
>             contextDir: 'config'
>             sourcePath: 'manifests'
>       managementState: Managed
>     dashboard:
>       managementState: Managed
>     datasciencepipelines:
>       managementState: Removed
>     kserve:
>       managementState: Removed
>     modelmeshserving:
>       managementState: Removed
>     ray:
>       devFlags:
>         manifests:
>           - uri: 'https://api.github.com/repos/opendatahub-io/kuberay/tarball/master'
>             contextDir: 'ray-operator/config'
>             sourcePath: 'default'
>       managementState: Managed
>     workbenches:
>       managementState: Managed
> ```
> 
> Edit: Updated CR

That makes perfect sense, thanks Karel! Adjusted that now. Unsure if it's the best way of doing it but I confirmed the sed commands I used for replacing those fields worked locally.

ChristianZaccaria commented:
/pj-rehearse

ChristianZaccaria commented:
/pj-rehearse

ChristianZaccaria commented:
/pj-rehearse

ChristianZaccaria commented:
/pj-rehearse

ChristianZaccaria commented:
/pj-rehearse

sutaakar commented:
@ChristianZaccaria Can you adjust `oc logs -n openshift-operators deployment/codeflare-operator-manager > ${ARTIFACT_DIR}/codeflare-operator.log` to use `opendatahub` namespace? That should help figuring out what is the issue.

ChristianZaccaria commented:
/pj-rehearse

ChristianZaccaria commented:
/pj-rehearse

ChristianZaccaria commented:
/pj-rehearse

ChristianZaccaria commented:
/retest

ChristianZaccaria commented:
/pj-rehearse

ChristianZaccaria commented:
/pj-rehearse

ChristianZaccaria commented:
/pj-rehearse

ChristianZaccaria commented:
/pj-rehearse

ChristianZaccaria commented:
/pj-rehearse

ChristianZaccaria commented:
/pj-rehearse

ChristianZaccaria commented:
/pj-rehearse

ChristianZaccaria commented:
/pj-rehearse

ChristianZaccaria commented:
/pj-rehearse

ChristianZaccaria commented:
/pj-rehearse

sutaakar reviewed (approved):
/lgtm

ChristianZaccaria commented:
/pj-rehearse ack
